tmt/steps/report/reportportal.py

```diff
@@ -14,85 +14,124 @@
 
 @dataclasses.dataclass
 class ReportReportPortalData(tmt.steps.report.ReportStepData):
+
+    @staticmethod
+    def _flag_env_to_default(option: str, default: bool) -> bool:
+        env_var = 'TMT_PLUGIN_REPORT_REPORTPORTAL_' + option.upper()
+        if env_var not in os.environ:
+            return default
+        return bool(os.getenv(env_var) == '1')
+
+    @staticmethod
+    def _str_env_to_default(option: str, default: Optional[str]) -> Optional[str]:
```

Comment:
So, I think it's time for `@overload`. Because you have two methods with different return value types: if `default` is an actual string, `_str_env_to_default()` will always return a string, never `None` (either envvar exists, or default is returned). And it's possible to distinguish them on the typing level, with `@overload` defining variants:

```python
@overload
@staticmethod
def _str_env_to_default(option: str, default: None) -> Optional[str]:
    pass

@overload
@staticmethod
def _str_env_to_default(option: str, default: str) -> str:
    pass

@staticmethod
def _str_env_to_default(option: str, default: Optional[str]) -> str:
    env_var = 'TMT_PLUGIN_REPORT_REPORTPORTAL_' + option.upper()
    if env_var not in os.environ or os.getenv(env_var) is None:
        return default
    return str(os.getenv(env_var))
```

This should remove the need for `str()` calls below, and mypy should be able to realize that `_str_env_to_default('defect_type', "^TMT_.*")` returns `str` which matches the declared type of `exclude_variables`.